FIX-#2684: Add method level docs for Modin XGBoost (#2685)
modin-project/modin: docs/modin_xgboost.rst

Distributed XGBoost functionality is placed in ``modin.experimental.xgboost`` module.
``modin.experimental.xgboost`` provides a xgboost-like API for ``train`` and ``predict`` functions.

.. automodule:: modin.experimental.xgboost
  :members: train

> ``train`` has all arguments of ``xgboost.train`` function except for `evals_result`

Reviewer: ```suggestion
``train`` has all arguments of the ``xgboost.train`` function except for ``evals_result``
```

Author: Suggestion was applied.

> As you can see ``train`` has all arguments of ``xgboost.train`` function exclude the `evals_result`

Reviewer: ```suggestion
``train`` has all arguments of ``xgboost.train`` function except for `evals_result`
```

Author: fixed

parameter which is returned as part of function return value instead of argument.

.. automodule:: modin.experimental.xgboost
  :noindex:
  :members: predict

> ``predict`` is separate function similar to ``xgboost.Booster.predict`` with an additional argument

Reviewer: ```suggestion
``predict`` is similar to ``xgboost.Booster.predict`` with an additional argument, 
```

Author: Suggestion was applied.

> ``predict`` is separate function unlike ``xgboost.Booster.predict`` which uses an additional argument

Reviewer: ```suggestion
``predict`` is separate function similar to ``xgboost.Booster.predict`` with an additional argument
```

Author: fixed
